File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        if len(set(s))==len(s):
            return len(s)
        res=[0]
        for i in range(len(s)):
            temp = s[i]
            for j in range(i+1,len(s)):
                if s[j] in temp:
                    break
                temp+=(s[j])
            res.append(len(temp))

        return (max(res))

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> mymap;
        int i = 0;
        int j =0 ;
        int ans = 0;
        while(i<s.size() && j<s.size()){
            if (mymap.find(s[j]) ==mymap.end()){
                            mymap.insert(s[j++]);
            ans = max(j-i,ans);

            }
            else{
mymap.erase(s[i++]);
        }}
        return ans;
    }
};